2748. Number of Beautiful Pairs/solution.py

```diff
@@ -0,0 +1,39 @@
+class Solution:
+    # Given a list of nums, it returns the number of beautiful pairs. A pair of
+    # numbers is called "beautiful" if the greatest common divisor of the first
+    # digit of the first number and the last digit of the second number have a 
+    # greatest common divisor of 1, which is exactly what the second function 
+    # handles.
+    def countBeautifulPairs(self, nums: List[int]) -> int:
+        # define the number of beautiful pairs.
+        
+        # define an alternate nums list that's a tuple of the first and last
+        # digits of the number.
+        
+        # for each of the alternate nums...
+            
+            # get the first digit of that number.
+            
+            # for each of the alternate nums after that one...
+                
+                # get the last digit of the number.
+                
+                # if the digits have a gcd() of 1, increase the number of 
+                # beautiful pairs.
+        
+    # This is a replacement for the standard gcd(), or Greatest Common Divisor 
+    # function passed in two integers and returning the greatest divisor that 
+    # divides them both.
+    def gcd(self, a: int, b: int) -> int:
```

Comment:
Hmm you are probably correct making a GCD function here. I was too lazy to do that.